.github/workflows/validate-pr-actions.yaml

```diff
@@ -0,0 +1,108 @@
+---
+name: Validate Github Actions
+
+on:
+  workflow_call:
+    inputs:
+      image:
+        type: string
+        required: false
+  pull_request:
+    paths:
+      - actions/**
+      - tests/test_resources/mpyl_config.yml
+      - tests/test_resources/repository/changed_files.json
+      - tests/projects/job/**
+
+# If multiple commits are pushed simultaneously, only build the last one.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+
+  health-check:
+    name: ▶️ Run actions/health-check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/health-check/action.yaml
+
+      - uses: ./actions/health-check
+        with:
+          config-path: tests/test_resources/mpyl_config.yml
+
+  lint-projects:
+    name: ▶️ Run actions/lint-projects
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/lint-projects/action.yaml
+
+      - uses: ./actions/lint-projects
+        with:
+          config-path: tests/test_resources/mpyl_config.yml
+
+  run-plan:
+    name: ▶️ Run actions/run-plan
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: yq -i '.runs.image = strenv(IMAGE)' actions/run-plan/action.yaml
+
+      - uses: ./actions/run-plan
+        with:
+          environment: pull-request
+          deploy-target: PullRequest
+          version: a-random-version
+          changed-files-path: tests/test_resources/repository/changed_files.json
+          config-path: tests/test_resources/mpyl_config.yml
+
+  run-stage:
+    name: ▶️ Run actions/run-stage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.2.1
+
+      - name: Point action.yaml files to a new version
+        if: inputs.image != ''
+        env:
+          IMAGE: docker://${{ inputs.image }}
+        run: |-
+          yq -i '.runs.image = strenv(IMAGE)' actions/run-plan/action.yaml
+          yq -i '.runs.image = strenv(IMAGE)' actions/run-stage/action.yaml
+
+      # To run a stage we always need to create a run plan before
+      # TODO find out if it's possible to write a stub run plan in tests/ so that we can test actions/run-stage in
```

Comment:
We definitely can btw, write it to `.mpyl` on root level though

Reply:
My point is more that the run plan is a pickled file, so it's not trivial to keep it up to date with changes to this repo. So that's for now it's much simpler to call `actions/run-plan` (to generate a new run plan) before calling `actions/run-stage`.

Reply:
Hm, actually if you don't have a run_plan and call mpyl while passing a `changed_files_path` (to avoid git usage) it will just create it because the logic happens in the run properties construction which is always called at the beginning. It's the same logic as `build status` uses, just without the extra prints.

Reply:
True, however `actions/run-stage` does not have a `changed-files-path` parameter because it's not expected to recalculate a build plan 😉

Reply:
That way we'll be testing a flow that will never happen in our CI/CD pipelines.